executors/imap/executor.go

```diff
@@ -0,0 +1,264 @@
+}
+
+// Run execute TestStep of type exec
+func (Executor) Run(ctx context.Context, l *log.Entry, aliases venom.Aliases, step venom.TestStep) (venom.ExecutorResult, error) {
+
+	var t Executor
+	if err := mapstructure.Decode(step, &t); err != nil {
+		return nil, err
+	}
+
+	start := time.Now()
+
+	if l.Level == log.DebugLevel {
+		imapLogMask = imap.LogConn | imap.LogState | imap.LogCmd
+		imapSafeLogMask = imap.LogConn | imap.LogState
+	}
+
+	result := Result{Executor: t}
+	find, errs := t.getMail(l)
+	if errs != nil {
+		result.Err = errs.Error()
+	}
+	if find != nil {
+		result.Subject = find.Subject
+		result.Body = find.Body
+	} else {
+		result.Err = "searched mail not found"
+	}
+
+	elapsed := time.Since(start)
+	result.TimeSeconds = elapsed.Seconds()
+	result.TimeHuman = fmt.Sprintf("%s", elapsed)
+	result.Executor.IMAPPassword = "****hide****" // do not output password
+
+	return dump.ToMap(result, dump.WithDefaultLowerCaseFormatter())
+}
+
+func (e *Executor) getMail(l *log.Entry) (*Mail, error) {
+
+	if e.SearchFrom == "" && e.SearchSubject == "" {
+		return nil, fmt.Errorf("You have to use searchfrom and/or searchsubject")
+	}
+
+	c, errc := connect(e.IMAPHost, e.IMAPUser, e.IMAPPassword)
+	if errc != nil {
+		return nil, fmt.Errorf("Error while connecting:%s", errc.Error())
+	}
+
+	var box string
+
+	if e.MBox == "" {
+		box = "INBOX"
+	}
+
+	count, err := queryCount(c, box)
+	if err != nil {
+		disconnect(c)
+		return nil, fmt.Errorf("Error while queryCount:%s", err.Error())
+	}
+
+	l.Debugf("count messages:%d", count)
+
+	if count == 0 {
+		return nil, fmt.Errorf("No message to fetch")
+	}
+
+	messages, err := fetch(c, box, count, l)
+	if err != nil {
+		disconnect(c)
+		return nil, fmt.Errorf("Error while feching messages:%s", err.Error())
+	}
+
+	for _, msg := range messages {
+		m, erre := extract(msg, l)
+		if erre != nil {
+			return nil, erre
+		}
+
+		if e.isSearched(m) {
+			if e.MBoxIfSuccess != "" {
+				l.Debugf("Move to %s", e.MBoxIfSuccess)
+				if err := m.move(c, e.MBoxIfSuccess); err != nil {
+					return nil, err
+				}
+			}
+			return m, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Mail not found")
+}
+
+func (e *Executor) isSearched(m *Mail) bool {
+	if e.SearchFrom != "" && e.SearchSubject != "" {
+		return strings.Contains(m.From, e.SearchFrom) ||
+			strings.Contains(m.Subject, e.SearchSubject)
+	}
+	if e.SearchFrom != "" {
+		return strings.Contains(m.From, e.SearchFrom)
+	}
+	if e.SearchSubject != "" {
+		return strings.Contains(m.Subject, e.SearchSubject)
+	}
+
+	return false
+}
+
+func (m *Mail) move(c *imap.Client, mbox string) error {
+	seq, _ := imap.NewSeqSet("")
+	seq.AddNum(m.UID)
+
+	if _, err := c.UIDMove(seq, mbox); err != nil {
+		return fmt.Errorf("Error while move msg to %s, err:%s", mbox, err.Error())
+	}
+	return nil
+}
+
+func connect(host, imapUsername, imapPassword string) (*imap.Client, error) {
+
+	c, errd := imap.DialTLS(host+":993", nil)
```

Comment:
port should be an option, or should be appended only if `host` doesn't contain any `:`